### PR TITLE
Added sidebar message for no description

### PR DESCRIPTION
**Pull request by Malorn44**
Should work. Couldn't find any more cases of the error I was getting where it randomly wasn't working as expected.

### CODE REVIEW

File: web/src/app/listing/component.html [hunk 1]
```diff
@@ -16,19 +16,19 @@
...
 			</table>
-			<span *ngIf="showDescriptionTooltip" class="description-tooltip position-absolute" placement="right" [ngbPopover]="listing.description" container="body" triggers="mouseenter:mouseleave">
+			<span *ngIf="showDescriptionTooltip" class="description-tooltip position-absolute" placement="right" [ngbPopover]="tooltipDescription" container="body" triggers="mouseenter:mouseleave">
 				<img src="assets/images/info.svg" />
 			</span>
 			<span (click)="descriptionClick()" *ngIf="!showDescriptionTooltip" class="course-description" [class.hideDescription]="!showingDescription" [class.showDescription]="showingDescription" [class.selectedDescription]="isCourseSelected()">
 				{{listing.description}}
 			</span>
 			<div class="menu position-absolute" (click)="showingMenu=!showingMenu">
-				<div class = "arrow-container down-arrow" *ngIf="!showingMenu">
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: what's the reason for this change?

**Comment by Bad-Science** [comment]
Awesome! Thanks for the PR. Had one question, otherwise looks good.

File: web/src/app/listing/component.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import { ConflictsService } from '../services/conflicts.service'
 export class ListingComponent implements OnInit{
   @Input() listing: Listing;
   @Input() showDescriptionTooltip: boolean = false;
-  @Input() showDescription: boolean = false;
+  @Input() showDescription: boolean = false;
```
Comment (Bad-Science) [inline web/src/app/listing/component.ts]: remove extra space plox

File: web/src/app/listing/component.html (hunk 1 above)
Comment (Malorn44) [reply web/src/app/listing/component.html]: uhhh..... actually I'm not sure. I never changed any of these lines except 19 myself.

**Comment by Malorn44** [comment]
image
image
Some images to see what it does.

**Comment by Malorn44** [comment]
Updated to include a bootstrap alert message for issue #328

File: web/src/app/listing/component.html (hunk 1 above)
Comment (Bad-Science) [reply web/src/app/listing/component.html]: Please revert this change and I'll merge this :)

File: web/src/app/sidebar/interested-courses/component.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export class InterestedCoursesComponent implements OnInit {
     // display interested courses on sidebar
     // display message to try selecting some if none
     if (this.listingIds.size > 0) {
-      this.statusText = '';
+      document.getElementById('sidebar-stat').classList.add("hide-sidebar-status");
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.ts]: You shouldn't use the `document` selector functions without a very compelling reason. In general you should not be writing logic to edit the view in a controller. To conditionally display an element, in general use `*ngIf`.

File: web/src/app/sidebar/interested-courses/component.scss [hunk 4]
```diff
@@ -20,4 +20,13 @@
 	.sidebar-status {
 		min-width: 375px;
 	}
-}
\ No newline at end of file
+}
+
+.hide-sidebar-status {
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.scss]: see below, use `*ngIf`

File: web/src/app/listing/component.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <div class = "course" [class.fixed-height]="!showingMenu && !showingDescription" (mouseenter)="hovered=true" (mouseleave) = "hovered=false">
 	<div class="row flex-nowrap h-100 min-height">
 	  <div class="indicator-container" (click)="clickCourse()">
-	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered && !isCourseSelected()" [class.indicator-selected-hover]="hovered && isCourseSelected()"></div>
+	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered" ></div>
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: This looks like an unintentional change as well, please revert.

File: web/src/app/listing/component.html [hunk 6]
```diff
@@ -16,19 +16,19 @@
 		    	<div class="course-details">{{creditRange}}</div>
 		    </tr>
 			</table>
-			<span *ngIf="showDescriptionTooltip" class="description-tooltip position-absolute" placement="right" [ngbPopover]="listing.description" container="body" triggers="mouseenter:mouseleave">
+			<span *ngIf="showDescriptionTooltip" class="description-tooltip position-absolute" [placement]="['right','bottom','left','top','auto']" [ngbPopover]="tooltipDescription" container="body" triggers="mouseenter:mouseleave">
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: Tried this locally, working fine :+1:

**Comment by Malorn44** [comment]
updated to use ngif instead of document.getElementById

**Comment by Malorn44** [comment]
ok let me know if any other things need to be modified

File: web/src/app/listing/component.html (hunk 1 above)
Comment (Malorn44) [reply web/src/app/listing/component.html]: done

File: web/src/app/listing/component.html (hunk 5 above)
Comment (Malorn44) [reply web/src/app/listing/component.html]: done

File: web/src/app/listing/component.html [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <div class = "course" [class.fixed-height]="!showingMenu && !showingDescription" (mouseenter)="hovered=true" (mouseleave) = "hovered=false">
 	<div class="row flex-nowrap h-100 min-height">
 	  <div class="indicator-container" (click)="clickCourse()">
-	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered && !isCourseSelected()" [class.indicator-selected-hover]="hovered && isCourseSelected()"></div>
+	  	<div class="indicator h-100" [class.indicator-selected]="isCourseSelected()" [class.indicator-unselected]="!isCourseSelected()" [class.indicator-hover]="hovered" && !isCourseSelected()" [class.indicator-selected-hover]="hovered && isCourseSelected()"></div>
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: There is still an unintentional change here, extra `"`

File: web/src/app/listing/component.html [hunk 8]
```diff
@@ -26,7 +26,7 @@
 				<div class = "arrow-container down-arrow" *ngIf="!showingMenu">
 					<i class="arrow down"></i>
 				</div>
-				<div class="arrow-container" *ngIf="showingMenu">
+				<div class = "arrow-container *ngIf="showingMenu">
```
Comment (Bad-Science) [inline web/src/app/listing/component.html]: Still an unintentional change, missing quote and added spaces

File: web/src/app/sidebar/interested-courses/component.html [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <div class="container">
     <div class="sidebar-status">
-    	{{statusText}}
+    	<div *ngIf="showStatusText" class="alert alert-sidebar"> Try selecting some courses :) </div>
```
Comment (Bad-Science) [inline web/src/app/sidebar/interested-courses/component.html]: :+1:
